File: push_swap_prototype/internal/generator/generator.go
package generator

import (
	"math/rand"
	"time"
)

// GenerateRandomNumbers generates an array of random numbers from 0 to size-1
func GenerateRandomNumbers(size int) []int {
	rand.Seed(time.Now().UnixNano())
	
	// Create array with numbers 0 to size-1
	numbers := make([]int, size)
	for i := 0; i < size; i++ {
		numbers[i] = i
	}
	
	// Shuffle the array
	for i := len(numbers) - 1; i > 0; i-- {
		j := rand.Intn(i + 1)
		numbers[i], numbers[j] = numbers[j], numbers[i]
	}
	
	return numbers
}

File: push_swap_prototype/lab/src/utils/generator.c
#include <stdlib.h>
#include <time.h>

int * generate_random_array(int size)
{
    int *array = malloc(size * sizeof(int));
    
    // Seed the random number generator
    srand(time(NULL));
    
    // Initialize array with numbers 0 to size-1
    for (int i = 0; i < size; i++)
    {
        array[i] = i;
    }
    
    // Fisher-Yates shuffle to randomize the order
    for (int i = size - 1; i > 0; i--)
    {
        int j = rand() % (i + 1);
        // Swap elements
        int temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }
    
    return array;
}